### ISSUE TITLE
Black screen

### ISSUE THREAD

**Issue by Benzillaist** [dnschneid/crouton#2923]
```
Please paste the output of the following command here: sudo edit-chroot -all
```
name: trusty
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/trusty...
crouton: version 1-20161129162558~master:5f2f11f9
release: trusty
architecture: amd64
xmethod: xorg
targets: xfce,extension
host: version 9000.10.0 (Official Build) dev-channel kip 
kernel: Linux localhost 3.10.18 #1 SMP Mon Nov 28 21:55:59 PST 2016 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Not unmounting /mnt/stateful_partition/crouton/chroots/trusty as another instance is using it.
#### Please describe your issue:
When i go into chroot i get a black screen

**Comment by bradley2W1DL** [comment]
I'm having the same issue where the screen blacks out immediately after running the `sudo enter-chroot startxfce` command.
```
Please paste the output of the following command here: sudo edit-chroot -all
```
name: precise
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/precise...
crouton: version 1-20161129162558~master:5f2f11f9
release: precise
architecture: amd64
xmethod: xorg
targets: xfce
host: version 9000.10.0 (Official Build) dev-channel parrot 
kernel: Linux localhost 3.8.11 #1 SMP Mon Nov 28 22:00:25 PST 2016 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/precise...

__Please describe your issue:__

When i go into chroot i get a black screen

__If known, describe the steps to reproduce the issue:__

`sudo startxfce4` immediately blacks out the screen.  If I then wait for a time it seems that booting up the chroot fails and it returns to the chronos shell and you can see the output below.  (sorry about the big log dump, I'm not sure what info in there might be pertinent.)

```
Entering /mnt/stateful_partition/crouton/chroots/precise...
/usr/bin/startxfce4: Starting X server

_XSERVTransmkdir: ERROR: euid != 0,directory /tmp/.X11-unix will not be created.

X.Org X Server 1.15.1
Release Date: 2014-04-13
X Protocol Version 11, Revision 0
Build Operating System: Linux 2.6.42-75-generic x86_64 Ubuntu
Current Operating System: Linux localhost 3.8.11 #1 SMP Mon Nov 28 22:00:25 PST 2016 x86_64
Kernel command line: cros_secure  console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-1 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="2 vboot none ro 1,0 2579200 bootcache PARTUUID=b1f9b33e-6651-4c48-a561-9998857a9233/PARTNROFF=1 2579200 f036231c4b22c766c9b8aa3b91ad4e65fbb3cf0f 512 20000 100000, vroot none ro 1,0 2539520 verity payload=254:0 hashtree=254:0 hashstart=2539520 alg=sha1 root_hexdigest=c38a68ff3c7f607a2a3702c1e957da15160d70af salt=045402b1c948afbcefd7e78ae054b4653c0ac134487c0ccee8968b42f8125034" noinitrd vt.global_cursor_default=0 kern_guid=b1f9b33e-6651-4c48-a561-9998857a9233 add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic iTCO_vendor_support.vendorsupport=3  
Build Date: 12 February 2015  03:37:52PM
xorg-server 2:1.15.1-0ubuntu2~precise5 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.30.2
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Sat Dec  3 08:50:48 2016
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
setversion 1.4 failed: Permission denied
Initializing built-in extension Generic Event Extension
Initializing built-in extension SHAPE
Initializing built-in extension MIT-SHM
Initializing built-in extension XInputExtension
Initializing built-in extension XTEST
Initializing built-in extension BIG-REQUESTS
Initializing built-in extension SYNC
Initializing built-in extension XKEYBOARD
Initializing built-in extension XC-MISC
Initializing built-in extension SECURITY
Initializing built-in extension XINERAMA
Initializing built-in extension XFIXES
Initializing built-in extension RENDER
Initializing built-in extension RANDR
Initializing built-in extension COMPOSITE
Initializing built-in extension DAMAGE
Initializing built-in extension MIT-SCREEN-SAVER
Initializing built-in extension DOUBLE-BUFFER
Initializing built-in extension RECORD
Initializing built-in extension DPMS
Initializing built-in extension X-Resource
Initializing built-in extension XVideo
Initializing built-in extension XVideo-MotionCompensation
Initializing built-in extension XFree86-VidModeExtension
Initializing built-in extension XFree86-DGA
Initializing built-in extension XFree86-DRI
Initializing built-in extension DRI2
Loading extension GLX
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)
setversion 1.4 failed: Permission denied
crouton: version 1-20161129162558~master:5f2f11f9
release: precise
architecture: amd64
xmethod: xorg
targets: xfce
host: version 9000.10.0 (Official Build) dev-channel parrot 
kernel: Linux localhost 3.8.11 #1 SMP Mon Nov 28 22:00:25 PST 2016 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
/etc/xdg/xfce4/xinitrc: 1: /etc/xdg/xfce4/xinitrc: ssh-agent: not found
_IceTransmkdir: ERROR: euid != 0,directory /tmp/.ICE-unix will not be created.

(xfce4-session:4143): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfsettingsd:4150): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfwm4:4148): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
xfdesktop[4157]: starting up

(xfdesktop:4157): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
xfce4-settings-helper: Another instance is already running. Leaving...

(xfce4-panel:4156): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(wrapper:4185): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(polkit-gnome-authentication-agent-1:4173): GLib-CRITICAL **: g_variant_new_string: assertion `string != NULL' failed

(polkit-gnome-authentication-agent-1:4173): polkit-gnome-1-WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files
running 'ssh-agent -s -k'
/etc/xdg/xfce4/xinitrc: 1: eval: ssh-agent: not found
Running exit commands...
/usr/bin/xinit: connection to X server lost

waiting for X server to shut down xscreensaver: 08:51:32: SIGHUP received: restarting...
Hangup
xscreensaver: 08:51:32: running as bird/bird (1000/1000)

xscreensaver: 08:51:33: Can't open display: :1
xscreensaver: 08:51:33: running as bird/bird (1000/1000)

xscreensaver: 08:51:33: Errors at startup are usually authorization problems.
              But you're not logging in as root (good!) so something
              else must be wrong.  Did you read the manual and the FAQ?

              http://www.jwz.org/xscreensaver/faq.html
              http://www.jwz.org/xscreensaver/man.html

(EE) Server terminated successfully (0). Closing log file.

Unmounting /mnt/stateful_partition/crouton/chroots/precise...
```

**Comment by dvwz** [comment]
I'm having the same issue.

**Comment by dcecile** [comment]
If I'm having the same issue... after _rebooting_ one of my old chroots continued to work fine, while my newly-updated chroot failed to start X properly. Before rebooting, both chroots were fine.

It sounds like this was triggered by a recent Chrome OS change?

**Comment by dcecile** [comment]
*Update*: OK, I tried this again, and I don't think that a Chrome OS change came with my update.

That's because I was able to get X working by *closing my laptop to put it to sleep, then trying again*. For my laptop, something is getting changed during the wake-up or the sleep that suddenly "enables" X to start working for me.

If I try starting X immediately after booting up my laptop, it won't work. But after a sleep + wake-up, X starts working without problems.

@Benzillaist: Maybe you can try this out to see if my problem is the same as yours?

**Comment by bjascob** [comment]
I'm having the same issue with precise.  It had been working fine for weeks but after I did a "restart to update" on the chrome power button, the X server quit starting when doing sudo startxfce4.  I'm assuming that something changed in the latest update to the chrome-os but I haven't figured out what yet.  Please post if you have any ideas on what to try.

**Comment by DennisLfromGA** [comment]
@dcecile,

> That's because I was able to get X working by closing my laptop to put it to sleep, then trying again. For my laptop, something is getting changed during the wake-up or the sleep that suddenly "enables" X to start working for me.

That's an interesting find.
I hope others will try it and report back with their results.

-**DennisL**

**Comment by ScriptMonkey00** [comment]
Same problem happened to me... Powered down and saw the chromebook "I'm going to update" icon in the corner.  Now it's on and when I start my trusty lxde chroot, I just get a blank/black screen.

I did try closing the lid and opening it again as mentioned earlier, and that didn't seem to help.  Any information or updates about this would be greatly appreciated.  I"m not sure it matters, but here's my log of edit-chroot all:

name: trustylxde
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/trustylxde...
crouton: version 1-20161129162558~master:5f2f11f9
release: trusty
architecture: amd64
xmethod: xorg
targets: keyboard,lxde,audio
host: version 9000.29.0 (Official Build) beta-channel wolf 
kernel: Linux localhost 3.8.11 #1 SMP Wed Dec 14 21:56:31 PST 2016 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/trustylxde...

**Comment by bjascob** [comment]
Closing and opening the lid for me didn't work but I decided to try xiwi.  Doing an upgrade (something like sudo sh crouton -u -r trusty -t xiwi) and installing the xiwi extension to chrome worked to get things running again.  Of course I'm using the xiwi target now, not xorg but I don't see a big difference in what I'm doing.

**Comment by ScriptMonkey00** [comment]
Posting that edit-chroot dump reminded me I was on beta channel (from back when crouton worked on beta but not stable)... I did a backup, then hopped back to stable channel and restored and I'm back working again!  Let's hope that this glitch in the beta channel doesn't make it to stable.

**Comment by DennisLfromGA** [comment]
*Just throwing this out there in hopes that it might be of help.*

Some withe the same or similar issue have resolved it using this method: https://github.com/dnschneid/crouton/issues/2917#issuecomment-267411158

*Hope this helps,*
-**DennisL**

**Comment by zelemo** [comment]
Just to let people know - I've started having this problem, and the method above (2917 comment) solved this issue for me.  Acer chromebook, using startxkde.

**Comment by aou** [comment]
while tracking down another issue (corrupted glyphs when using intel drivers, whether from ubuntu or 01.org), I tried editing `/etc/crouton/xorg-intel-sna.conf` to:
```
Section "Device"
   Identifier "Intel Graphics SNA+TearFree"
   Driver     "intel"
   Option     "AccelMethod" "uxa"
   # Option     "TearFree"    "true"
EndSection
```

basically changing the driver to the older uxa accelerator. corrupted font glyphs disappeared, but as an added bonus, xfce now loads via -X xorg!

**Comment by aou** [comment]
Actually, one advantage of `uxa` over `modesetting` is, steam, glxgears, etc, will still run.

**Comment by CauseOfExceptions** [comment]
Hey guys just as a note, I was also having this issue on trusty xfce. Other apps were also misbehaving on my chromebook so I switched it back to "stable" channel ver 55.0.2883.103. I was on v56, development channel before. I reinstalled my chroot and now it works as before, no problems. There seems to be an issue with that chrome OS version.

**Comment by Benzillaist** [comment]
Hello! For everyone with/without the IO error you can:

1. Log out
2. do ctrl+shift+alt+r Twice, and you will be prompted to revert back to the previous version. 
3. wait...
4. wait...
5. so much more waiting that the old narrator quit and they had to hire a new one.
6. Do the basic process

PLEASE NOTE: THIS WILL CLEAR ALL OF YOUR DATA ON YOUR CHROMEBOOK!!!

I hope this helps,

Ben

**Comment by dbjungle** [comment]
I can confirm I got this problem on v56.0.2924.53 from the beta channel as well.

**Comment by pmarkkula** [comment]
@aou 

> Actually, one advantage of uxa over modesetting is, steam, glxgears, etc, will still run.

Could you clarify what you mean by being able to run those applications? I am using modesetting driver in Trusty and I am able to run glxgears.

**Comment by aou** [comment]
@pmarkkula 

at least on my machine (dell chromebook 13), back when i tried it last, setting the driver and accelmethod to modesetting did not enable hardware acceleration.

are you using both driver and accel method set to modesetting?

**Comment by pmarkkula** [comment]
This is what I am using at the moment:
```
Section "Device"
   Identifier "Modesetting"
   Driver     "modesetting"
   Option     "AccelMethod" "glamor"
   Option     "PageFlip"    "true"
EndSection
```
I am also using Xenial graphics stack. That might make a difference.

**Comment by MegaWalls** [comment]
how to edit `/etc/crouton/xorg-intel-sna.conf`?

**Comment by MegaWalls** [comment]
`(trusty)thomas@localhost:~$ /usr/share/X11/xorg.conf.d
-su: /usr/share/X11/xorg.conf.d: Is a directory`
that is the stage that I got to. How can I edit the file to `Section "Device"
   Identifier "Intel Graphics SNA+TearFree"
   Driver     "intel"
   Option     "AccelMethod" "uxa"
   # Option     "TearFree"    "true"
EndSection
`

**Comment by drinkcat** [comment]
Bissecting on samus.

- Chrome OS 8997.0.0 is good
- Chrome OS 8998.0.0 is bad (bad means: master branch black screen, _and_ nointeldrv branch crashes the kernel)

There is nothing of interest in Chrome OS-land (https://crosland.corp.google.com/log/8997.0.0..8998.0.0: google-only link).

Chromium, as usual, has tons of commits: https://chromium.googlesource.com/chromium/src/+log/56.0.2920.0..56.0.2923.0?pretty=fuller&n=10000

There _are_ some drm/GPU stuff in there.

I'll try to bisect Chromium tomorrow.

**Comment by drinkcat** [comment]
Probably linked with this issue: https://bugs.chromium.org/p/chromium/issues/detail?id=655770, and related patch https://chromium.googlesource.com/chromium/src.git/+/547665586e0d17adc9d8ecf03b41c9448fe7719c .

Workaround here: https://github.com/dnschneid/crouton/pull/3081

**Comment by dnschneid** [comment]
Keeping this open for a real fix

**Comment by GhostZodick** [comment]
Will we have such issue every time Chrome OS updates?

**Comment by D0nvt** [comment]
Not sure if this helps at all, but I noticed this was a problem when I switched to the Beta channel in December. I moved back to the Stable channel then, but now that that build has moved to stable there's no way to avoid it. I'll continue trying some different target mixes and post again if I find one that seems to work.

**Comment by e1337sheep** [comment]
The workaround [#3081](https://github.com/dnschneid/crouton/pull/3081) - how do us common "users" use this workaround?

**Comment by dmnorth** [comment]
Just updating the crouton in question worked for me. As in:
sudo sh ~/Downloads/crouton -u -n chrootname

**Comment by D0nvt** [comment]
@dmnorth can confirm updating chroot resolved the issue. I'm running XFCE on trusty.

**Comment by casnacaj** [comment]
I can confirm that update of chroot works for trusty but it does not work for xenial.

**Comment by drinkcat** [comment]
@casnacaj : Can you please provide the output of croutonversion?

**Comment by nemo-nullius** [comment]
Can confirm that update of trusty works now (three days ago it still did not work)! (trusty+i3/xfce/lxde) Thanks!

**Comment by rtpavlovsk21** [comment]
@drinkcat If you think you can work around this i would suggest looking at [#2926](https://github.com/dnschneid/crouton/issues/2926#issuecomment-270139802). I have had this issue on other linux systems as well (I fixed a MATE desktop installation with a similar set of paramters). I am not sure of the extent (or if I have truly found the source) but I changed this on recommendation listed in [Arch's xf86-video-intel](https://wiki.archlinux.org/index.php/intel_graphics#X_freeze.2Fcrash_with_intel_driver) wiki. On some of the [Arch Forums](https://www.reddit.com/r/archlinux/comments/4cojj9/it_is_probably_time_to_ditch_xf86videointel/) libcg thinks that this driver is undersupported.

I havent had a chance to try your fix, if it doesn't work I'll try to give you as much info as possible.

**Comment by casnacaj** [comment]
@drinkcat 
`crouton: version 1-20170216202114~master:2f7cfb5e
release: xenial
architecture: amd64
xmethod: xorg
targets: chrome,x11,extension,keyboard
host: version 9000.82.0 (Official Build) stable-channel reks 
kernel: Linux localhost 3.18.0-13527-gc2f2230 #1 SMP PREEMPT Wed Feb 1 15:06:00 PST 2017 x86_64 x86_64 x86_64 GNU/Linux
freon: yes`

**Comment by casnacaj** [comment]
However I updated today and it works now ;-). Thank you very much guys!

**Comment by drinkcat** [comment]
@rtpavlovsk21 did you try updating your chroot? I think the work around should work in most (all?) cases, but please let us know if it doesn't.

xf86-video-modesetting has other issues that lead to a kernel crash, ever since that Chromium commit, that cause the blank screen issue with xf86-video-intel driver. But, yes, we probably want to look at that too.

**Comment by rtpavlovsk21** [comment]
@drinkcat Your fix does work. I'm not really an expert on any of this, so the modesetting workarounds were sufficient for the several weeks that this problem was happening. 

If there is a claim about a race condition I don't know how I would verify, deny or test to make sure that the issue is put to bed... but the condition seems to be dependent on the driver in my case because I was able to use crouton for several months w/ acceleration per @pmarkkula 's comments.

Thanks for your efforts!
```bash
Entering /mnt/stateful_partition/crouton/chroots/xenial...
Creating crouton group with GID 1001...
Exiting... run in root env
Creating unassociated wayland group...
Exiting... run in root env
Exiting... run in root env
Exiting... run in root env
/home/pavlovsr
sudo: no tty present and no askpass program specified

chgrp: invalid group: ‘crouton’
_XSERVTransmkdir: Owner of /tmp/.X11-unix should be set to root

X.Org X Server 1.18.4
Release Date: 2016-07-19
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.4.0-45-generic x86_64 Ubuntu
Current Operating System: Linux localhost 3.8.11 #1 SMP Mon Feb 6 22:19:56 PST 2017 x86_64
Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 2539520 verity payload=PARTUUID=27f60fb5-ddc8-ed49-83b2-6ae3e7d36fdb/PARTNROFF=1 hashtree=PARTUUID=27f60fb5-ddc8-ed49-83b2-6ae3e7d36fdb/PARTNROFF=1 hashstart=2539520 alg=sha1 root_hexdigest=720374f5371c71cd5b9e08ac172ebed1045f7a5f salt=48417ae7a1449bf43026d73aee64b4860a079b54bd5e37d7da9f081b41bb13e6" noinitrd vt.global_cursor_default=0 kern_guid=27f60fb5-ddc8-ed49-83b2-6ae3e7d36fdb add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic iTCO_vendor_support.vendorsupport=3  
Build Date: 02 November 2016  10:06:10PM
xorg-server 2:1.18.4-0ubuntu0.2 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.33.6
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Fri Feb 17 19:59:01 2017
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)
Error getting authority: Error initializing authority: Could not connect: No such file or directory
crouton: version 1-20170216202114~master:2f7cfb5e
release: xenial
architecture: amd64
xmethod: xorg
targets: lxde
host: version 9202.18.0 (Official Build) dev-channel peppy 
kernel: Linux localhost 3.8.11 #1 SMP Mon Feb 6 22:19:56 PST 2017 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
chgrp: invalid group: ‘crouton’
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
Kicking xrandr again
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
Kicking xrandr again
Agent pid 7938
** Message: main.vala:99: Session is LXDE
** Message: main.vala:100: DE is LXDE
** Message: main.vala:131: log directory: /home/pavlovsr/.cache/lxsession/LXDE
** Message: main.vala:132: log path: /home/pavlovsr/.cache/lxsession/LXDE/run.log
(II) AIGLX: Suspending AIGLX clients for VT switch
```

**Comment by dnschneid** [comment]
@drinkcat out of curiosity, does the xrandr poke workaround avoid the modesetting crash as well, or does it happen before we can tickle the display back to life?

**Comment by drinkcat** [comment]
The crash happens quite quickly so I doubt the xrandr poke would work. Didn't try, though.

**Comment by w-barath** [comment]
I notice it's trying to configure eDP1 but after the update the display name (in xrandr) has changed to eDP-0 so maybe you should change your script to try that display instead.

**Comment by rkhtsmj** [comment]
This edit to /etc/crouton/xorg-intel-sna.conf stabled my **-r trusty -t lxde,xorg,xiwi,keyboard** so xorg is actually usable now:

<Section "Device"
   Identifier "Intel Graphics SNA+TearFree"
   Driver     "intel"
#   Option      "AccelMethod" "sna"
#   Option      "TearFree"    "true"
    Option      "AccelMethod" "uxa"
EndSection>

Ironically, of course, the file name is no longer appropriate. :) Does anyone know where X11 references the /etc/crouton directory? Thanks for the hack!!!
